anime_downloader/sites/animesvostfr.py

```diff
@@ -0,0 +1,262 @@
+            # 'photo': 'https://hydrax.net/watch?v=wZsRPKrmb'
+            #
+            # Sample 2
+            # 'rapidvideo': 'https://youtubedownloader.cx/v/1e660ijq873-ldr'
+            # 'photoss': 'https://play.comedyshow.to/embedplay/d187decc7e7596c0c9ad831c4a09038d'
+            # 'opencdn': 'https://lb.cartoonwire.to/public/dist/index.html?id=77790c2132de4b173c86db105a96b5af'
+            # 'photos': 'https://hydrax.net/watch?v=pVkQjLHxO'
+            # 'photo': 'https://hydrax.net/watch?v=pVkQjLHxO'
+            #
+            # Sample 3
+            # 'rapidvideo': ''
+            # 'photoss': 'https://play.comedyshow.to/embedplay/b6bab9c780325ff53644cbc4dba32558'
+            # 'opencdn': 'https://lb.cartoonwire.to/public/dist/index.html?id=cb5c81f15593ae88cf17bebe0f73b177'
+            # 'photos': 'https://hydrax.net/watch?v=rTaZnpvMdp'
+            # 'photo': 'https://hydrax.net/watch?v=rTaZnpvMdp'
+            #
+            # Sample 4
+            # 'rapidvideo': 'https://youtubedownloader.cx/v/xd33pu5401xy78p'
+            # 'photoss': 'https://play.comedyshow.to/embedplay/1c047309aebdb11797aa93f0b689eb9f'
+            # 'opencdn': 'https://lb.cartoonwire.to/public/dist/index.html?id=No m3u8Id'
+            # 'photos': ''
+            # 'photo': ''
+            #
+            #
+            urls_server = {} 
+            for serv in self.SERVERS:
+                urls_server[serv] = helpers.get(self.link_stream_url, params={'server' : serv, 'filmId' : filmid}).text.strip()
+               
+            logger.debug( "_get_sources urls: {}".format(urls_server) )
+            
+            sources = []
+  
+            for serv in self.SERVERS:
+                if urls_server[serv] and (urls_server[serv] > '') and (urls_server[serv].find('No m3u8Id') < 0):
+                
+                    soup = None
+                    try:
+                        soup = helpers.soupify(helpers.get(urls_server[serv]))
+                    except HTTPError as err:
+                        logger.debug("_get_sources HTTP Err: %s", str(err))
+                    
+                    if soup:
+                        #logger.debug( "soup: {}".format(soup) )
+                        iframe = soup.select_one("iframe")
+                        if iframe:
+                            url = iframe['src'].strip()
+                            if url and (url > ''):
+                                logger.debug( "%s -> %s", urls_server[serv], url )
+                                urls_server[serv] = url 
+                            
+                    #video = soup.select_one("video") 
+                    #if video:
+                    #    logger.debug( "_get_sources urls: {}".format(video) )
+                        
+                    
+                    #if serv == "rapidvideo":
+                    #    sources.append(('no_extractor', urls_server[serv]))
+                    if 'youtubedownloader' in urls_server[serv]:
```

Comment:
a for loop would be great here

Reply:
yes but i didn't modify config.py for the moment :p

Reply:
Doesn't seem relevant to config. You can replace the lines with something like this
```
servers = {
'fembed':'gcloud'
} #etc....
for a in servers:
	if a in urls_server[serv]:
		sources.append({'extractor':servers[a], 'server':a, 'url':urls_server[serv], 'version':'subbed'})
```

Reply:
i have to do it but i need that also :

```
# TODO: put it into config.py   
            if not self.sitename in Config['siteconfig']:         
                Config['siteconfig'][self.sitename] = {
                    "servers": [
                        "youtubedownloader",
                        "gcloud",
                        "feurl",
                        "fembed",
                        "hydrax"
                    ]
                }

```
because self.sort_sources(sources) need that and do something like:

```
servers_extractor = {
'fembed':'gcloud'
} #etc....
for a in Config['siteconfig'][self.sitename]["servers"]:
	if a in urls_server[serv]:
		sources.append({'extractor': servers_extractor[a], 'server':a, 'url':urls_server[serv], 'version':'subbed'})
```

Reply:
if you're going to do that, use `self.config` instead of `Config['siteconfig'][self.sitename]["servers"]`

Reply:
Yes it's already done in the last "final" push
But i think i could create a function like : FindBetterExtractor( url )
and modify it again and maybe we could add it into the base AnimeEpisode Class ?